src/tokens/BlessCurseManager.ttslua

```diff
@@ -1,139 +1,101 @@
-        color = CURSE_COLOR
-    })
+function onSave() return JSON.encode(TOGGLE) end
 
-    self.createButton({
-        label = "Return",
-        click_function = "returnCurseToken",
-        function_owner = self,
-        position = { 2.1, 0.1, 0.5 },
-        height = 150,
-        width = 400,
-        scale = { x = 1.75, y = 1.75, z = 1.75 },
-        font_size = 100,
-        font_color = { r = 1, g = 1, b = 1 },
-        color = CURSE_COLOR
-    })
+function onLoad(saved_state)
+    if saved_state ~= nil then
+        TOGGLE = JSON.decode(saved_state)
+    else
+        TOGGLE = false
+    end
 
-    self.createButton({
-        label = "Reset", click_function = "doReset", function_owner = self,
-        position = { 0, 0, 1.8 }, rotation = { 0, 0, 0 }, height = 350, width = 800,
-        font_size = 250, color = { 0, 0, 0 }, font_color = { 1, 1, 1 }
-    })
+    -- index: 0 - bless
+    buttonParamaters.click_function = "clickBless"
+    buttonParamaters.position       = { -1.03, 0.05, 0.46 }
+    self.createButton(buttonParamaters)
+
+    -- index: 1 - curse
+    buttonParamaters.click_function = "clickCurse"
+    buttonParamaters.position[1]    = -buttonParamaters.position[1]
+    self.createButton(buttonParamaters)
+
+    -- index: 2 - alternative mode (take / return)
+    buttonParamaters.click_function = "enableAlt"
+    buttonParamaters.width          = 900
+    buttonParamaters.height         = 210
+    buttonParamaters.position       = { -1.03, 0.05, -0.85 }
+    self.createButton(buttonParamaters)
+
+    -- index: 3 - default mode (add / remove)
+    buttonParamaters.click_function = "enableDefault"
+    buttonParamaters.position[1]    = -buttonParamaters.position[1]
+    self.createButton(buttonParamaters)
+
+    -- load labels, tooltips and colors
+    updateButtons()
+
+    -- context menu
+    self.addContextMenuItem("Remove all", doRemove)
+    self.addContextMenuItem("Reset", doReset)
+    self.addContextMenuItem("More Information", function()
+        printToAll("------------------------------", "White")
+        printToAll("Bless / Curse Manager v" .. information["version"] .. " by Chr1Z", "Orange")
+        printToAll("last updated: " .. information["last_updated"], "White")
+        printToAll("original by Tikatoy", "White")
+    end)
+
+    -- hotkeys
+    addHotkey("Bless Curse Status", printStatus, false)
+    addHotkey("Wendy's Menu", addMenuOptions, false)
 
+    -- initializing tables
     numInPlay = { Bless = 0, Curse = 0 }
     tokensTaken = { Bless = {}, Curse = {} }
     sealedTokens = {}
     Wait.time(initializeState, 1)
 
-    addHotkey("Bless Curse Status", printStatus, false)
-    addHotkey("Wendy's Menu", addMenuOptions, false)
+    -- feedback for token arranger
+    if tokenarranger then
+        print("Bless/Curse Manager will update the Token Arranger automatically.")
+        print("If you remove the Token Arranger, save and reload before continuing.")
```

Comment:
Can we add something to the Token Arranger (probably in onDestroy()) to make this unnecessary?

Reply:
done